nixos/modules/services/networking/pangolin.nix

```
@@ -0,0 +1,536 @@
+              "~CAP_SYS_PACCT"
+              "~CAP_SYS_PTRACE"
+              "~CAP_SYS_TIME"
+              "~CAP_SYS_TTY_CONFIG"
+              "~CAP_SYSLOG"
+              "~CAP_WAKE_ALARM"
+            ];
+            ProtectSystem = "full";
+            ProtectHome = true;
+            PrivateTmp = "disconnected";
+            PrivateDevices = true;
+            PrivateMounts = true;
+            ProtectKernelTunables = true;
+            ProtectKernelModules = true;
+            ProtectKernelLogs = true;
+            ProtectControlGroups = true;
+            LockPersonality = true;
+            RestrictRealtime = true;
+            ProtectClock = true;
+            ProtectProc = "noaccess";
+            ProtectHostname = true;
+            NoNewPrivileges = true;
+            RestrictSUIDSGID = true;
+            MemoryDenyWriteExecute = true;
+            RestrictAddressFamilies = [
+              "AF_INET"
+              "AF_INET6"
+              "AF_NETLINK"
+              "AF_UNIX"
+            ];
+            SystemCallFilter = [
+              "~@aio:EPERM"
+              "~@chown:EPERM"
+              "~@clock:EPERM"
+              "~@cpu-emulation:EPERM"
+              "~@debug:EPERM"
+              "~@keyring:EPERM"
+              "~@memlock:EPERM"
+              "~@mount:EPERM"
+              "~@obsolete:EPERM"
+              "~@pkey:EPERM"
+              "~@privileged:EPERM"
+              "~@raw-io:EPERM"
+              "~@reboot:EPERM"
+              "~@resources:EPERM"
+              "~@sandbox:EPERM"
+              "~@setuid:EPERM"
+              "~@swap:EPERM"
+              "~@sync:EPERM"
+              "~@timer:EPERM"
+            ];
+
+            ExecStart = utils.escapeSystemdExecArgs [
+              (lib.getExe pkgs.fosrl-gerbil)
+              "--reachableAt=http://localhost:${toString cfg.gerbilPort}"
+              "--generateAndSaveKeyTo=${toString cfg.dataDir}/config/key"
+              "--remoteConfig=http://localhost:${toString finalSettings.server.internal_port}/api/v1/gerbil/get-config"
+            ];
+          };
+        };
+        # make sure traefik places plugins in local dir instead of /
+        traefik.serviceConfig.WorkingDirectory = "${cfg.dataDir}/config/traefik";
+      };
+    };
+
+    services.traefik = {
+      enable = true;
+      group = "fossorial";
+      dataDir = "${cfg.dataDir}/config/traefik";
+      environmentFiles = lib.optional (!isNull cfg.traefikEnvironmentFile) cfg.traefikEnvironmentFile;
+      staticConfigOptions = {
+        providers.http = {
+          endpoint = "http://localhost:${toString finalSettings.server.internal_port}/api/v1/traefik-config";
+          pollInterval = "5s";
+        };
+        experimental.plugins.badger = {
+          moduleName = "github.com/fosrl/badger";
+          version = "v1.2.0";
+        };
+        certificatesResolvers.letsencrypt.acme =
+          (
+            if (finalSettings.domains.domain1.prefer_wildcard_cert) then
```

```suggestion
            if finalSettings.domains.domain1.prefer_wildcard_cert then
```

Useless parenthesis. Run `nixf-diagnose`. (which will soon be part of the treefmt in Nixpkgs)